### PR TITLE
Add quadrature rules

### PR DESCRIPTION
**Pull request by ktpr1223214**

### CODE REVIEW

File: recipe/quadrature/quadrature.cc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#include "recipe/quadrature/quadrature.h"
+#include <cassert>
+
+namespace recipe {
+namespace quadrature {
+
+double TrapezoidalRule(double (*f)(double), const double a, const double b,
```
Comment (i05nagai) [inline recipe/quadrature/quadrature.cc:7]: It is safe to put assert for a case of `num_of_partition <= 0` or add a comment to documents of this function.

File: recipe/quadrature/quadrature.cc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  double integral = (f(b) + f(a)) / 2.0;
+
+  for (int i = 1; i < num_of_partition; i++) {
+    integral += f(a + i * h);
+  }
+
+  return h * integral;
+}
+
+double SimpsonRule(double (*f)(double), const double a, const double b,
```
Comment (i05nagai) [inline recipe/quadrature/quadrature.cc:21]: It is safe to put assert for a case of `num_of_partition <= 0` or add a comment to documents of this function.

File: recipe/quadrature/quadrature.cc [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#include "recipe/quadrature/quadrature.h"
+#include <cassert>
+
+namespace recipe {
+namespace quadrature {
```
Comment (i05nagai) [inline recipe/quadrature/quadrature.cc:5]: Maybe we can use `integrate` for a more generic namespace. There are a couple of algorithms which is not categorized as `quadrature` such as `monte-carlo` and `quasi monte-carlo`. Then `quadrature` can be sub-namespace of it.

**Review by i05nagai — APPROVED** [review]
Happy to have documented code 👍

**Comment by ktpr1223214** [comment]
Thanks your comments.

**Pull request merged**